src/Blocks/components/form/assets/form.js

```diff
@@ -1658,6 +1661,46 @@ export class Form {
 		this.utils.setOnFocus(event.target);
 	};
 
+	/**
+	 * On keypress event for regular fields.
+	 *
+	 * @param {object} event Event callback.
+	 *
+	 * @returns {void}
+	 */
+	onKeyDownEvent = (event) => {
+		const formId = this.state.getFormIdByElement(event.target);
+		const field = this.state.getFormFieldElementByChild(event.target);
+		const name = field.getAttribute(this.state.getStateAttribute('fieldName'));
+
+		if (this.state.getStateElementTypeCustom(name, formId) === 'number') {
+			const allowedKeys = [
+				'Backspace',
+				'Enter',
```

Comment:
Can't you just filter things when setting state/submitting the form?
This seems a bit hacky and could maybe be bypassed with assistive technologies etc

Reply:
the problem is that I don't have anything to filter as if you get the value with text the value will be empty 👎 and I can't filter anything :(((